src/tableau_server/tableau_server_objects.py

```diff
@@ -0,0 +1,372 @@
+        """
+            Set the attribute to an integer,
+            if the class has the attribute,
+            and if the attribute is a string.
+        """
+        if self.__existing_str_attr(attr):
+            setattr(self, attr, int(getattr(self, attr)))
+
+    def __to_bool(self, attr: str):
+        """
+            Set the attribute to a boolean,
+            if the class has the attribute,
+            and if the attribute is a string.
+        """
+        if self.__existing_str_attr(attr):
+            setattr(self, attr, getattr(self, attr).lower() == 'true')
+
+    def __post_init__(self):
+        # Convert to Datetime
+        self.__to_datetime('created_at')
+        self.__to_datetime('completed_at')
+        self.__to_datetime('last_login')
+        self.__to_datetime('next_run_at')
+        self.__to_datetime('updated_at')
+        # Convert to Boolean
+        self.__to_bool('attach_image')
+        self.__to_bool('attach_pdf')
+        self.__to_bool('content_send_if_view_empty')
+        self.__to_bool('data_acceleration_config_acceleration_enabled')
+        self.__to_bool('embed_password')
+        self.__to_bool('encrypt_extracts')
+        self.__to_bool('has_alert')
+        self.__to_bool('has_extracts')
+        self.__to_bool('is_certified')
+        self.__to_bool('is_published')
+        self.__to_bool('is_embedded')
+        self.__to_bool('query_tagging_enabled')
+        self.__to_bool('show_tabs')
+        self.__to_bool('suspended')
+        self.__to_bool('top_level_project')
+        self.__to_bool('use_remote_query_agent')
+        self.__to_bool('writeable')
+        # Convert to Integer
+        self.__to_int('connected_workbooks_count')
+        self.__to_int('contents_counts_datasource_count')
+        self.__to_int('contents_counts_project_count')
+        self.__to_int('contents_counts_view_count')
+        self.__to_int('contents_counts_workbook_count')
+        self.__to_int('favorites_total')
+        self.__to_int('finish_code')
+        self.__to_int('port')
+        self.__to_int('priority')
+        self.__to_int('server_port')
+        self.__to_int('sheet_count')
+        self.__to_int('size')
+        self.__to_int('usage_total_view_count')
+        self.__to_int('user_count')
+
+    def __hash__(self):
+        return hash(str(astuple(self)))
+
+    def to_dict(self):
+        return asdict(self)
+
+
+@dataclass
+class User(ServerObject):
+    """ A Tableau User object """
+    name: str
+    full_name: str = None
+    domain_name: str = None
+    auth_setting: str = None
+    email: str = None
+    email_domain: str = None
+    external_auth_user_id: str = None
+    site_role: str = None
+    locale: str = None
+    language: str = None
+    last_login: datetime = None
+
+    def __post_init__(self):
+        if not self.email_domain and self.email:
+            self.email_domain = self.email.lower().split('@')[-1]
+        super().__post_init__()
+
+    def __hash__(self):
```

Comment:
what are all these hashes used for?

Reply:
This dunder method allows classes to be identified as unique.
It's useful if you wanted to two your classes into a `set()` for example:
```python
obj1 = ServerObject()
obj2 = ServerObject()
obj3 = ServerObject()
unique_objs = {obj1, obj2, obj3}
'''
Returns => { ServerObject() }
'''
```